src/components/MainView/MainView.jsx

```diff
@@ -47,66 +50,137 @@ export const MainView = () => {
     localStorage.clear();
   };
 
-  const handleMovieClick = (movie) => {
-    setSelectedMovie(movie);
+  const handleUpdateUser = (updatedUser) => {
```

Comment:
The methods `handleDeregisterUser` and `handleUpdateUser` should also call the appropriate API endpoints.